pkg/reconciler/v1alpha1/revision/resources/deploy_test.go

```diff
@@ -39,12 +39,219 @@ import (
+		}},
+	}
+
+	defaultFluentdContainer = &corev1.Container{
+		Name:      FluentdContainerName,
+		Image:     "indiana:jones",
+		Resources: fluentdResources,
+		Env: []corev1.EnvVar{{
+			Name:  "FLUENTD_ARGS",
+			Value: "--no-supervisor -q",
+		}, {
+			Name:  "SERVING_CONTAINER_NAME",
+			Value: UserContainerName,
+		}, {
+			Name: "SERVING_CONFIGURATION",
+			// No owner reference
+		}, {
+			Name:  "SERVING_REVISION",
+			Value: "bar",
+		}, {
+			Name:  "SERVING_NAMESPACE",
+			Value: "foo",
+		}, {
+			Name: "SERVING_POD_NAME",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.name",
+				},
+			},
+		}},
+		VolumeMounts: fluentdVolumeMounts,
+	}
+
+	defaultPodSpec = &corev1.PodSpec{
+		Volumes:                       []corev1.Volume{varLogVolume},
+		TerminationGracePeriodSeconds: refInt64(45),
+	}
+
+	defaultDeployment = &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "foo",
+			Name:      "bar-deployment",
+			Labels: map[string]string{
+				serving.RevisionLabelKey: "bar",
+				serving.RevisionUID:      "1234",
+				AppLabelKey:              "bar",
+			},
+			Annotations: map[string]string{},
+			OwnerReferences: []metav1.OwnerReference{{
+				APIVersion:         v1alpha1.SchemeGroupVersion.String(),
+				Kind:               "Revision",
+				Name:               "bar",
+				UID:                "1234",
+				Controller:         &boolTrue,
+				BlockOwnerDeletion: &boolTrue,
+			}},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &one,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					serving.RevisionUID: "1234",
+				},
+			},
+			ProgressDeadlineSeconds: &ProgressDeadlineSeconds,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						serving.RevisionLabelKey: "bar",
+						serving.RevisionUID:      "1234",
+						AppLabelKey:              "bar",
+					},
+					Annotations: map[string]string{
+						sidecarIstioInjectAnnotation: "true",
+					},
+				},
+				// Spec: filled in by makePodSpec
+			},
+		},
+	}
 )
 
 func refInt64(num int64) *int64 {
 	return &num
 }
 
+type containerOpt func(*corev1.Container)
+type podSpecOpt func(*corev1.PodSpec)
+type deploymentOpt func(*appsv1.Deployment)
+
+func makeTestContainer(defaultContainer *corev1.Container, opts ...containerOpt) *corev1.Container {
+	container := defaultContainer.DeepCopy()
+	for _, opt := range opts {
+		opt(container)
+	}
+	return container
+}
+
+func makeTestUserContainer(opts ...containerOpt) *corev1.Container {
+	return makeTestContainer(defaultUserContainer, opts...)
+}
+
+func makeTestQueueContainer(opts ...containerOpt) *corev1.Container {
+	return makeTestContainer(defaultQueueContainer, opts...)
+}
+
+func makeTestFluentdContainer(opts ...containerOpt) *corev1.Container {
+	return makeTestContainer(defaultFluentdContainer, opts...)
+}
+
+func setContainerEnvVar(name, value string) containerOpt {
```

Comment:
Should we follow the naming "convention" laid out in https://github.com/knative/serving/blob/master/pkg/reconciler/v1alpha1/testing/functional.go, where functional options are prefixed with `with`?

In this case for example that could be

```golang
func withEnvVar(name, value string) containerOpt {
...
}
```

Maybe we could move these options to `functional.go` even as they are quite general purpose?

Reply:
Thanks for the pointer, the new names read much better.

If there are more places these can be used today, or may be in the future, I agree they should be available in functional.go.

I can create an issue for myself to move these out to functional.go as a follow up.